### PR TITLE
feat: search feature implemented

### PR DESCRIPTION
**Pull request by JowelTisso**
This PR is for search implementation

### CODE REVIEW

File: src/components/header/Header.jsx [hunk 1]
```diff
@@ -35,6 +39,13 @@ const Header = () => {
     sidenavDispatch({ type: TOGGLE_NAV });
   };
 
+  const applySearch = ({ key, target }) => {
+    if (key === "Enter") {
+      const filteredBySearch = filterByTitle(target.value, products);
+      filterDispatch({ type: PRODUCT_DATA, payload: filteredBySearch });
```
Comment (Tarandeep-s1ngh) [inline src/components/header/Header.jsx:47]: you can pass payload as object for better readability and prophylaxis

**Review by aniketxparihar — APPROVED** [review]
Looks Good

**Review by Tarandeep-s1ngh — APPROVED** [review]
The search bar is visible on the home/landing page also, but it is not working, you can either remove it from there or redirect the user to the products page on searching from the home page.


Everything else looks great!

- Clean File structure
- Dividing the code into multiple components
- Good naming conventions

Good job!

**Review by manismk — APPROVED** [review]
The functionality works properly. Looks good

File: src/components/header/Header.jsx (hunk 1 above)
Comment (JowelTisso) [reply src/components/header/Header.jsx:47]: Ok, will try that

**Pull request merged**